src/main/scala/mrpowers/jodie/ChangeDataFeedHelper.scala

```diff
@@ -0,0 +1,276 @@
+package mrpowers.jodie
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
+import org.apache.spark.sql.delta.actions.{AddCDCFile, Metadata}
+import org.apache.spark.sql.delta.commands.cdc.CDCReader
+import org.apache.spark.sql.delta.commands.cdc.CDCReader.isCDCEnabledOnTable
+import org.apache.spark.sql.delta.util.FileNames
+import org.apache.spark.sql.delta.{DeltaLog, VersionNotFoundException}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import java.io.IOException
+import scala.collection.mutable
+import scala.util.control.Breaks.{break, breakable}
+
+object ChangeDataFeedHelper {
+  def apply(path: String, startingVersion: Long): ChangeDataFeedHelper = {
+    val deltaLog = DeltaLog.forTable(SparkSession.active, path)
+    new ChangeDataFeedHelper(path, startingVersion, deltaLog.snapshot.version, deltaLog)
+  }
+
+  def apply(path: String, startingVersion: Long, endingVersion: Long): ChangeDataFeedHelper = {
+    val deltaLog = DeltaLog.forTable(SparkSession.active, path)
+    new ChangeDataFeedHelper(path, startingVersion, endingVersion, deltaLog)
+  }
+}
+
+/** *
+ * Helper Class and methods for working with different failure scenarios while using Change Data Feed provided by OSS Delta Lake
+ * Detailed explanation under : [[https://medium.com/@joydeep.roy/change-data-feed-failure-scenarios-recovery-explained-5606c65d0c2e]]
+ *
+ * @param path
+ * @param startingVersion
+ * @param endingVersion
+ * @param deltaLog
+ */
+case class ChangeDataFeedHelper(path: String, startingVersion: Long, endingVersion: Long, deltaLog: DeltaLog) {
```

Comment:
As of now, I have intentionally kept the design to work with versions as it applies to the issues described in the blog. Let me explore if this can be done for timestamps as well. I will probably raise a separate PR for the same